shared-test/uk/gov/hmrc/thirdpartyorchestrator/utils/ApplicationBuilder.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.thirdpartyorchestrator.utils
+
+import java.time.LocalDateTime
+
+import uk.gov.hmrc.apiplatform.modules.applications.domain.models._
+import uk.gov.hmrc.apiplatform.modules.common.domain.models._
+import uk.gov.hmrc.apiplatform.modules.submissions.domain.models.SubmissionId
+import uk.gov.hmrc.thirdpartyorchestrator.domain.models.applications._
+
+trait ApplicationBuilder {
+
+  def buildApplication(applicationId: ApplicationId, clientId: ClientId, userId: UserId, submissionId: SubmissionId, name: String, createdOn: LocalDateTime): Application = {
+    Application(
+      applicationId,
```

Comment:
Named arguments make it easier to read the builders. When you set something to `None` or 18 without the argument name, it needs something more than just reading the code, e.g., hovering the mouse or opening the case class side-by-side.

Reply:
Done